// ember-test-addon/index.js
'use strict';

const fs = require('fs');
const Plugin = require('broccoli-plugin');
const mergeTrees = require('broccoli-merge-trees');
const path = require('path');

AddCustomFile.prototype = Object.create(Plugin.prototype);
AddCustomFile.prototype.constructor = AddCustomFile;
function AddCustomFile(inputNodes, options) {
  this.path = options.path;
  Plugin.call(this, inputNodes);
}

AddCustomFile.prototype.build = function() {
  var outPath = path.join(this.outputPath, this.path);
  fs.mkdirSync(path.dirname(outPath));
  fs.writeFileSync(outPath, '// File for test tree imported and added via postprocessTree()');
};

var customTestFilePath = 'my-custom-test-assets/test-dependency.js';

module.exports = {
  name: require('./package').name,

  included(app) {
    this._super.included(app);

    app.import(customTestFilePath, { type: 'test' });
  },

  postprocessTree(type, tree){
    if (type === 'test') {
      return mergeTrees([tree, new AddCustomFile([tree], {
        path: customTestFilePath
      })]);
    }
    return tree;
  }
};
