// fake-template-linter/index.js
'use strict';

const fs = require('fs-extra');
const path = require('path');
const Plugin = require('broccoli-plugin');
const walkSync = require('walk-sync');

module.exports = {
  name: require('./package').name,

  lintTree(type, tree) {
    if (type === 'templates') {
      return new TemplateLinter(tree, {
        annotation: 'TemplateLinter'
      });
    }
  }
};

function TemplateLinter (inputNode, options) {
  Plugin.call(this, [inputNode], {
    annotation: options.annotation
  });

  this.options = options;
}
TemplateLinter.prototype = Object.create(Plugin.prototype);
TemplateLinter.prototype.constructor = TemplateLinter;

TemplateLinter.prototype.build = function () {
  var basePath = this.inputPaths[0];
  var templateFiles = walkSync(basePath)
        .filter(function(entry) {
          return /\.hbs$/.test(entry);
        });

  for (var i = 0; i < templateFiles.length; i++) {
    var templateFilePath = templateFiles[i];

    var contents = fs.readFileSync(path.join(basePath, templateFilePath), { encoding: 'utf8' });
    var testContents = 'QUnit.test("TemplateLint: ' + templateFilePath + '", function(assert) {\n' +
          'assert.ok(false, "DO NOT USE BADJUJU");\n' +
          '})';
    var testFilePath = path.join(this.outputPath, templateFilePath + '-test.js');

    if (/BADJUJU/.test(contents)) {
      fs.mkdirsSync(path.dirname(testFilePath));

      fs.writeFileSync(testFilePath, testContents, { encoding: 'utf8' });
    }
  }
};
